tools/infrasys/src/main.rs

```diff
@@ -0,0 +1,872 @@
+            ..Default::default()
+        })
+        .await
+        .context(error::CreateStack { stack_name, region })?;
+    // We don't have to wait for successful stack creation to grab the stack ARN
+    let stack_arn = stack_result
+        .clone()
+        .stack_id
+        .context(error::ParseResponse {
+            what: "stack_id",
+            resource_name: stack_name,
+        })?;
+
+    // Grab the StackOutputs to get the Bucketname and BucketURL
+    let output_array = get_stack_outputs(&cfn_client, &stack_name, region).await?;
+    let bucket_name = output_array[0]
+        .output_value
+        .as_ref()
+        .context(error::ParseResponse {
+            what: "outputs[0].output_value (bucket name)",
+            resource_name: stack_name,
+        })?
+        .to_string();
+    let bucket_url = output_array[1]
+        .output_value
+        .as_ref()
+        .context(error::ParseResponse {
+            what: "outputs[1].output_value (bucket url)",
+            resource_name: stack_name,
+        })?
+        .to_string();
+
+    Ok((stack_arn, bucket_name, bucket_url))
+}
+
+/// Adds a BucketPolicy allowing GetObject access to a specified VPC
+/// Input: Region and name of bucket, which folder the Policy should be scoped to, and vpcid
+/// Note that the prefix parameter must have the format "/<folder>/*" and the bucket name "<name>"
+/// Output: Doesn't need to save any metadata from this action  
+async fn add_bucket_policy(
+    region: &String,
+    bucket_name: &String,
+    prefix: &String,
+    vpcid: &String,
+) -> Result<()> {
+    // Get old policy
+    let s3_client =
+        S3Client::new(Region::from_str(region).context(error::ParseRegion { what: region })?);
+    let mut current_bp = match s3_client
+        .get_bucket_policy(GetBucketPolicyRequest {
+            bucket: bucket_name.clone(),
+            expected_bucket_owner: None,
+        })
+        .await
+    {
+        Ok(output) => {
+            let mut policy = output.policy.context(error::ParseResponse {
+                what: "policy",
+                resource_name: bucket_name,
+            })?;
+            policy.insert_str(policy.len() - 2, &(",".to_string()));
+            policy
+        }
+        Err(..) => "{\"Version\": \"2008-10-17\",
+                     \"Statement\": []}"
+            .to_string(),
+    };
+
+    // Create a new policy
+    // QUESTION: string insertion is fickle :(
+    let new_bucket_policy = format!(
+        "{{
+                       \"Effect\": \"Allow\",
+                        \"Principal\": \"*\",
+                        \"Action\": \"s3:GetObject\",
+                        \"Resource\": \"arn:aws:s3:::{}{}/*\",
+                        \"Condition\": {{
+                            \"StringEquals\": {{
+                                \"aws:sourceVpce\": \"{}\"
+                            }}
+                        }}
+                    }}",
+        bucket_name, prefix, vpcid
```

Comment:
I think instead this string manipulation we're doing, (which i agree is really fickle and dangerous), I would recommend modifying the bucket policy via [serde_json::value](https://docs.serde.rs/serde_json/value/index.html). Which should make it cleaner and safer. 

It should technically be possible to insert a `serde_json::value` into another `serde_json::value`.
The process would be to first deserialize the policy string out to a `serde_json::value`, then get a mutable reference to `policy["Statements"]` (Which should be an array of `serde_json::value`s. See https://docs.serde.rs/serde_json/enum.Value.html#method.as_array_mut), then serialize the `new_bucket_policy` into a separate `serde_json::value` and push that into the policy.statements array.  Then finally we can serialize out the new policy as a json string.